experimental/WreathMacdonaldpols/src/WreathMacdonaldpols.jl

```diff
@@ -0,0 +1,256 @@
+# I would like to credit Dario Mathiä who produced an initial version of the following code
+
+import Chevie: complex_reflection_group, charinfo, CharTable, CyclotomicNumbers, Cyc
+export wreath_macs
+
+# Tools
+
+#gives indices of a subarray
+function subarray_indices(sub,arr)
+  res = Int[]
+  for i in 1:length(sub)
+    push!(res,findfirst(x->x==sub[i],arr))
+  end
+  return res
+end
+
+# Computes the b-invariant of a partition
+function b1(lambda::Vector{Int64})
+  if length(lambda)==0
+    return 0
+  end
+  return sum(i->(i-1)*lambda[i],1:length(lambda))
+end
+
+# Computes the b-invariant of a multipartition
+function btot(lbb::Multipartition)
+  return b1(map(lambda -> sum(lambda)[1], lbb)) + length(lbb)*sum(i -> b1(data(lbb[i])),1:length(lbb))
+end
+
+function beta_number_to_partition(beta::Vector{Integer})
+  lb=Integer[]
```

Comment:
```suggestion
function beta_number_to_partition(beta::Vector{Int})
  lb = Int[]
```
vectors of abstract types (like here `Integer`) have poor performance in julia. If possible, use a concrete type here. Either `Int` if all occurring numbers are at most 2^63-1, or `ZZRingElem` otherwise.

This also applies to all other occurrences of `Integer` in this file

Reply:
Thank you very much for that. I didn't knew. I will change all occurrences of Integer to Int.